src/pages/distroSettings/tabs/ProviderTab/transformers.ts

```diff
@@ -29,13 +44,24 @@ export const formToGql = ((data, distro) => {
   } = data;
 
   switch (providerName) {
-    case Provider.Static: {
+    case Provider.Static:
+      return {
+        ...distro,
+        provider: Provider.Static,
+        providerSettingsList: [
+          ...staticProviderSettings(data.providerSettings).gql,
+        ],
+      };
+    case Provider.Docker:
       return {
         ...distro,
-        provider: providerName,
-        ...staticProviderSettings(data.providerSettings).gql,
+        provider: Provider.Docker,
+        providerSettingsList: [
+          ...dockerProviderSettings(data.providerSettings).gql,
+        ],
+        // @ts-ignore-error - containerPoolId will exist in DockerFormState.
```

Comment:
We can narrow down the type here using the `in` operator to avoid this error ([docs](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing)). Something like this:
```js
    case Provider.Docker: {
      if ("containerPoolId" in data.providerSettings) {
        return {
          ...distro,
          provider: Provider.Docker,
          providerSettingsList: [
            ...dockerProviderSettings(data.providerSettings).gql,
          ],
          containerPool: data.providerSettings.containerPoolId,
        };
      }
      break;
    }
```